services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/doi/DoiManagerImpl.java

```diff
@@ -111,6 +112,7 @@ DoiAssociation createOrUpdateAssociation(DoiAssociation dto) throws RecoverableM
 			association = doiAssociationDao.updateDoiAssociation(dto);
 		} catch (NotFoundException e1) { // The DOI does not already exist (exception was thrown by getEtag)
 			try {
+				dto.setAssociatedBy(dto.getUpdatedBy());
 				association = doiAssociationDao.createDoiAssociation(dto); // Create
```

Comment:
In dao, catch IllegalArgument on duplicate key and throw DuplicateKeyException (maybe from Spring, possibly make new)